### ISSUE TITLE
Change SA1121 to trigger on alias statement instead of alias use

### ISSUE THREAD

**Issue by bjornhellander** [DotNetAnalyzers/StyleCopAnalyzers#3640]
Consider this example:
```
using Cookie = System.Int32;

namespace TestNamespace
{
    class TestClass
    {
        Cookie x; // SA1121 triggers here

        void TestMethod(Cookie y) // And here
        {
            Cookie z; // And here...
        }
    }
}
```

I find it very surprising that SA1121 is reported where the alias is used instead of where it is defined. If I have a file where I really want to have an alias, then having the diagnostic reported on the alias statement instead of on every single use of the alias would make the experience much more pleasant. As it stands now, my only option seems to be to suppress the rule in the entire file. That way, I lose the benefit of the rule everywhere in the file, just because I have one single Int32 alias that I have decided to use. I think this is very unfortunate.

This change might also make the analyzer cheaper performance wise.
